core/src/main/scala/sttp/tapir/server/interceptor/log/ServerLogInterceptor.scala

```diff
@@ -22,10 +19,10 @@ class ServerLogInterceptor[T, F[_]](log: ServerLog[T], toEffect: (T, ServerReque
         decodeHandler
           .onDecodeSuccess(ctx)
           .flatMap { response =>
-            toEffect(log.requestHandled(ctx.endpoint, response.code.code), ctx.request).map(_ => response)
+            monad.unit(log.requestHandled(ctx.endpoint, ctx.request, response)).map(_ => response)
```

Comment:
This won't work unfortunately if the result of `log.requestHandled` is a lazily-evaluated side effect. The log will never get logged